descriptors/ModuleDescriptor-template.json

```diff
@@ -33,6 +33,17 @@
         }
       ]
     },
+    {
+      "id": "item-storage-batch-sync",
+      "version": "1.0",
+      "handlers": [
+        {
+          "methods": ["POST"],
+          "pathPattern": "/item-storage/sync",
```

Comment:
I think this path is ambiguous. The `sync` part could correspond to a synchronisation process rather than a synchronous operation.

Why does the path need to explain that this operation is synchronous, is it to avoid conflicts with the existing batch API or to create space for a future asynchronous variant?

The path does not include any mention that it is for a batch, why is that? Is it to avoid issues with the existing batch API?

cc: @skoczko

Reply:
"sync" is used to differentiate from an async API that may be introduced at a later stage.
While it is true that both synchronisation and synchronous have the same abbreviation sync this is not ambiguous because there is no synchronisation endpoint. The path name is used to distinguish existing endpoints, it doesn't replace the API documentation.

The GET /item-storage/items API also returns a batch of records without mentioning it in the path name and there exists POST /item-storage/items with the same path where only a single record can be uploaded in one request. Therefore the path is short and does not necessarily contain the complete documentation.

Reply:
> "sync" is used to differentiate from an async API that may be introduced at a later stage.
> While it is true that both synchronisation and synchronous have the same abbreviation sync this is not ambiguous because there is no synchronisation endpoint. The path name is used to distinguish existing endpoints, it doesn't replace the API documentation.

I personally find it ambiguous and confusing.

I would prefer something like `/item-storage/items/batch/synchronous` as I think it expresses it intent more clearly.

Please feel free to disregard this feedback given that you disagree with it.

> The GET /item-storage/items API also returns a batch of records without mentioning it in the path name 

I would suggest that the collection level get returns a collection or subset of records, not a batch.

I think of a batch as set of records submitted for processes, rather than any endpoint that involves multiple records.

> and there exists POST /item-storage/items with the same path where only a single record can be uploaded in one request. 

As I understand it, that is a specific convention that the collection endpoints are plural and creation occurs at the collection endpoint.

> Therefore the path is short and does not necessarily contain the complete documentation.

Agreed, I think it can be a reasonable indicator as to the intent of the endpoint.